flink/src/test/java/org/apache/iceberg/flink/SimpleDataUtil.java

```diff
@@ -191,9 +190,19 @@ public static void assertTableRows(Table table, List<RowData> expected) throws I
 
   public static void assertTableRecords(Table table, List<Record> expected) throws IOException {
     table.refresh();
+
```

Comment:
Why do we need this change? is it because two sets don't match after we add the additional metadata column? but wouldn't `_pos` and `_file` also have this issue?

Reply:
Because the metadata column projection logic produces additional columns even when the `requiredSchema` doesn't contain them and that is why we use `StructLikeSet`.  The `_pos` column shows up only when positional deletes exist,  the `_deleted` marker shows up when any of the deletes exist.

The failed unit test contains only equality delete which produces only `_deleted` column, so it failed with `HashMultiSet` comparison. But when the unit test, for example `TestIcebergFilesCommitter.TestCommitTwoCheckpointsInSingleTxn`, contains a positional delete, the unit test fails as well due to it has the additional column `_pos`. The following patch for the unit test could test it.
```
-      DeleteFile deleteFile1 = writeEqDeleteFile(appenderFactory, "delete-file-1", ImmutableList.of(delete3));
+      DeleteFile deleteFile1 = writePosDeleteFile(appenderFactory,
+          "pos-delete-file-1",
+          ImmutableList.of(Pair.of(dataFile1.path(), 3L)));
```

Reply:
I also find this suspicious. Is the extra column in the _expected_ records or the table? I don't think that this PR should change the data produced by `IcebergGenerics.read(table).build()`.

Reply:
It is in _expected records_. The extra column is added in `DeleteFileter#fileProjection`.

```
private static Schema fileProjection(Schema tableSchema, Schema requestedSchema,
                                       List<DeleteFile> posDeletes, List<DeleteFile> eqDeletes) {
    ...
    // We add it to requiredIds, so that it exists in missingIds when requestedSchema doesn't contain it.
    if (!posDeletes.isEmpty()) {
      requiredIds.add(MetadataColumns.ROW_POSITION.fieldId());
    }
    ....


    // We append it at the end anyway.
    if (missingIds.contains(MetadataColumns.ROW_POSITION.fieldId())) {
      columns.add(MetadataColumns.ROW_POSITION);
    }

    return new Schema(columns);
  }
```